app/src/Controller/LdapController.php

```diff
@@ -0,0 +1,247 @@
+     *
+     * @return JsonResponse
+     */
+    public function getLdapEntries(
+        Client $ldap,
+        Request $request,
+        SerializerInterface $serializer
+    ): JsonResponse {
+
+        $query = $request->get('query', '(objectClass=*)');
+        $baseDn = $request->get('base', null);
+
+        $options = [];
+
+        $filters = json_decode($request->get('filters', '[]'), true);
+        if (!empty($filters)) {
+            $queryFilters = '';
+            foreach ($filters as $field => $value) {
+                switch ($field) {
+                    case 'dn':
+                        $queryFilters .= "($value)";
+                        break;
+
+                    default:
+                        $queryFilters .= "($field=$value)";
+                        break;
+                }
+            }
+            $query = "(&$queryFilters$query)";
+        }
+        // XXX Is there any way to ask LDAP server to sort results?
+        //$orders  = json_decode($request->get('orders', null), true);
+
+        // Attributes option.
+        $attributes = $request->get('attributes', ['dn']);
+        $options['filter'] = $attributes;
+
+        // Add query options.
+        $size = (int) $request->get('size', 0);
+        $max = (int) $request->get('max', 0);
+
+        $options['pageSize'] = $size;
+        $options['maxItems'] = $max;
+
+        $ldap->bind();
+
+        //XXX Verify If there is a need to throw a special error by try catch
+        $ldapEntries = $ldap->search($query, $baseDn, $options);
+        $total = count($ldapEntries);
+        $entries = array();
+
+        foreach ($ldapEntries as $key => $ldapEntry) {
+            $entries[$key]['dn'] = $ldapEntry->getDn();
+
+            $ldapDto = new LdapEntryDTO();
+            //XXX Verify if there is a need to throw a special error by try catch
+            $ldapEntry =  $ldapDto->serializeJpegPhoto($ldapEntry);
+
+            // Rely on filter option to filter attributes
+            $entries[$key]['attributes'] = $ldapEntry->getAttributes();
+        }
+
+        $entries = $serializer->normalize(
+            $entries,
+            Entry::class
+        );
+
+        return new JsonResponse([
+            'total' => $total,
+            'items' => $entries,
+        ]);
+    }
+
+    /**
+     * @Route("/api/ldap/{entry}", name="get_ldap_entry", methods={"GET"})
+     *
+     * @return JsonResponse
+     */
+    public function getLdapEntryByDn(
+        string $entry,
+        Client $ldap,
+        Request $request,
+        SerializerInterface $serializer
+    ): JsonResponse {
+        $jsonResponse = new JsonResponse();
+
+        $query = $request->get('query', '(objectClass=*)');
+        // TODO Add attributes option.
+        $options = $request->get('options', []);
+
+        if (empty($entry)) {
+            // TODO Return translated error message.
+            return $jsonResponse->create(null, 400);
```

Comment:
Also, absolutely DO NOT create a `$jsonResponse` to call a static method from `JsonResponse`!! And that goes for everywhere it's done.